File: Assets/Scripts_CSharp/InputHandler.cs
using UnityEngine;
using System.Collections;

public class InputHandler : MonoBehaviour 
{
	void Update () 
	{
		if(Input.GetMouseButton(0))
		{
			Ray ray = Camera.mainCamera.ScreenPointToRay(Input.mousePosition);
			RaycastHit rayCastHit;
			
			if(Physics.Raycast(ray.origin, ray.direction, out rayCastHit, Mathf.Infinity))
			{
				Door door = rayCastHit.transform.GetComponent<Door>();
				if(door)
				{
					door.PlayDoorAnim();
				}
			}
		}
	}
	
	void OnGUI()
	{
		 GUI.Label(new Rect(10, 10,	500, 20), "Click on the door to trigger the animations!");
	}
}

File: Assets/Scripts_Javascript/InputHandler.js
#pragma strict

function Update () 
{
	if(Input.GetMouseButton(0))
	{
		var ray : Ray = Camera.main.ScreenPointToRay(Input.mousePosition);
		var rayCastHit : RaycastHit;
		
		if(Physics.Raycast(ray.origin, ray.direction, rayCastHit, Mathf.Infinity))
		{
			var door : Door = rayCastHit.transform.GetComponent(typeof(Door));
			if(door)
			{
				door.PlayDoorAnim();
			}
		}
	}
}